lightning/src/ln/onion_utils.rs

```diff
@@ -481,6 +481,24 @@ pub(super) fn process_onion_failure<T: secp256k1::Signing, L: Deref>(
 			},
 		};
 
+		// The failing hop includes either the inbound channel to the recipient or the outbound channel
+		// from the current hop (i.e., the next hop's inbound channel).
+		let num_blinded_hops = path.blinded_tail.as_ref().map_or(0, |bt| bt.hops.len());
+		is_from_final_node = route_hop_idx + 1 == path.hops.len() && num_blinded_hops <= 1;
```

Comment:
Is my understanding correct that the last hop in `Path::hops` is the introduction node, if there is a `BlindedTail`? And that the introduction node is not included in `BlindedTail::hops`? If so, why do we do we use `<= 1` and not `== 0`?

Reply:
For 1-hop blinded paths, the last hop in `Path::hops` is the recipient. Added a comment